boskos/metrics/metrics.go

```diff
@@ -39,33 +40,39 @@ var (
 	promMetrics = prometheusMetrics{
 		BoskosState: map[string]map[string]prometheus.Gauge{},
 	}
-	resources common.ResTypes
+	resources, states common.ResTypes
+	defaultStates     = []string{
+		common.Busy,
+		common.Cleaning,
+		common.Dirty,
+		common.Free,
+		common.Leased,
+	}
 )
 
 func init() {
 	flag.Var(&resources, "resource-type", "comma-separated list of resources need to have metrics collected")
+	flag.Var(&states, "resource-state", "comma-separated list of states need to have metrics collected")
+	flag.Parse()
+	if len(states) == 0 {
+		states = defaultStates
+	}
 }
 
 func initMetrics() {
 	for _, resource := range resources {
-		promMetrics.BoskosState[resource] = map[string]prometheus.Gauge{
-			common.Free: prometheus.NewGauge(prometheus.GaugeOpts{
-				Name: fmt.Sprintf("boskos_%s_%s", strings.Replace(resource, "-", "_", -1), common.Free),
-				Help: fmt.Sprintf("Number of %s %s", common.Free, resource),
-			}),
-			common.Busy: prometheus.NewGauge(prometheus.GaugeOpts{
-				Name: fmt.Sprintf("boskos_%s_%s", strings.Replace(resource, "-", "_", -1), common.Busy),
-				Help: fmt.Sprintf("Number of %s %s", common.Busy, resource),
-			}),
-			common.Dirty: prometheus.NewGauge(prometheus.GaugeOpts{
-				Name: fmt.Sprintf("boskos_%s_%s", strings.Replace(resource, "-", "_", -1), common.Dirty),
-				Help: fmt.Sprintf("Number of %s %s", common.Dirty, resource),
-			}),
-			common.Cleaning: prometheus.NewGauge(prometheus.GaugeOpts{
-				Name: fmt.Sprintf("boskos_%s_%s", strings.Replace(resource, "-", "_", -1), common.Cleaning),
-				Help: fmt.Sprintf("Number of %s %s", common.Cleaning, resource),
-			}),
+		promMetrics.BoskosState[resource] = map[string]prometheus.Gauge{}
+		for _, state := range states {
+			promMetrics.BoskosState[resource][state] = prometheus.NewGauge(prometheus.GaugeOpts{
+				Name: fmt.Sprintf("boskos_%s_%s", strings.Replace(resource, "-", "_", -1), state),
+				Help: fmt.Sprintf("Number of %s %s", state, resource),
+			})
 		}
+		// Adding other state for metrics that are not captured with existing state
+		promMetrics.BoskosState[resource][common.Other] = prometheus.NewGauge(prometheus.GaugeOpts{
```

Comment:
add it from the resource-state flag

Reply:
Decided not to do that as it makes the code more confusing.